### PR TITLE
Added Docker & Tar deploys to Deploy landing page

### PR DESCRIPTION
**Pull request by cwarnermm**
Added the following topics to the Deploy landing page:
- preview Mattermost using Docker (incorporates https://github.com/mattermost/docs/pull/5517)
- deploy Mattermost for production use (incorporates https://github.com/mattermost/docs/pull/5522)
- deploy Mattermost using Omnibus (incorporates https://github.com/mattermost/docs/pull/5351)

Made minor updates to subsequent subsections for flow.

### CODE REVIEW

**Comment by cwarnermm** [comment]
@nab-77 - In the generated preview, these changes are available on the Deploy landing page.

**Comment by cwarnermm** [comment]
Outstanding docs note: Redirects are needed for both ``getting-started/light-install.html`` and ``install/setting-up-local-machine-using-docker.html``.

File: source/guides/deployment.rst [hunk 1]
```diff
@@ -3,24 +3,187 @@ Deploy Mattermost
...
+
+          To access your new Mattermost deploy via HTTPS, navigate to ``https://<YOUR_MM_DOMAIN>/`` in your browser.
+
+          To shut down your deployment:
+
+            .. code:: bash
+  
+                sudo docker-compose -f docker-compose.yml -f docker-compose.nginx.yml down
+    
+    .. tab:: From Tar
```
Comment (nab-77) [inline source/guides/deployment.rst:158]: There are a bunch of steps we need to include here. The full list is here

http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/5522/install/install-tar.html

**Comment by azigler** [comment]
I fixed the GitLab instructions on #5517 and did so here as well.

**Comment by cwarnermm** [comment]
@azigler @justinegeffen - Your input on this PR is most welcome. This PR combines updates from 3 separate PRs into a single page of information. 

It's understood that the current Deployment landing page is getting long (and arguably too long). Our goal with this PR is to deliver easy to find, simplified deployment details for 3 primary deployment approaches. Future steps include updates to the website to link to this content.

**Review by azigler — APPROVED** [review]
@cwarnermm I really like this change! I'm just wondering about the tab order at this point. Right now, this proposes Tar/Omnibus/Docker. Should we change this to Docker/Omnibus/Tar or otherwise put Docker first? My understanding is we want to promote Docker as the primary deploy method. If it's first, then it also becomes simpler to change the Quick Install Guide links in [mattermost-server/PR#19924](https://github.com/mattermost/mattermost-server/pull/19924) and [mattermost-webapp/PR#10137](https://github.com/mattermost/mattermost-webapp/pull/10137) to point to the new URL (https://docs.mattermost.com/guides/deployment.html) and drop the Docker install links entirely, because this PR rolls them all into one page. Is that a correct understanding?

**Comment by cwarnermm** [comment]
You raise an excellent point about tab order, @azigler -- particularly with regards to other in-flight work. @nab-77 - thoughts?

File: source/install/troubleshooting.rst [hunk 2]
```diff
@@ -30,6 +30,56 @@ Important notes
...
+.. code:: bash
+
+   sudo rm -rf ./volumes
+
+Postgres issues
+~~~~~~~~~~~~~~~
+
+You can change the Postgres username and/or password (recommended) in the ``.env`` file.
+
+TLS & NGINX issues
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst:62]:
```suggestion
TLS and NGINX issues
```

File: source/install/troubleshooting.rst [hunk 3]
```diff
@@ -30,6 +30,56 @@ Important notes
...
+~~~~~~~~~~~~~~~~~~
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst:63]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~
```

File: source/install/troubleshooting.rst [hunk 4]
```diff
@@ -30,6 +30,56 @@ Important notes
 - Any manipulation of the database must be done using the built in command line tools.
 - Start simple with the step-by-step install guides for your operating system.
 
+Deployment troubleshooting
+--------------------------
+
+Docker deployments
+~~~~~~~~~~~~~~~~~~
+
+If you're deploying the Mattermost server using Docker on an M1 Mac and encountering permission issues in the Docker container, `re-reate the required directories and set their permissions <https://docs.mattermost.com/guides/deployment.html#deploy-mattermost-for-production-use>`__, then skip the following command:
```
Comment (justinegeffen) [inline source/install/troubleshooting.rst:39]:
```suggestion
If you're deploying the Mattermost server using Docker on an M1 Mac and encountering permission issues in the Docker container, `re-create the required directories and set their permissions <https://docs.mattermost.com/guides/deployment.html#deploy-mattermost-for-production-use>`__, then skip the following command:
```

Comment (justinegeffen) [reply source/install/troubleshooting.rst:39]: At what point does the command that must be skipped occur and is it a command that has a Y/N/Skip option?

File: source/guides/deployment.rst [hunk 5]
```diff
@@ -3,24 +3,369 @@ Deploy Mattermost
...
+
+      6. Deploy Mattermost.
+
+          **Without using the included NGINX:**
+
+          .. code:: bash
+  
+                sudo docker-compose -f docker-compose.yml -f docker-compose.without-nginx.yml up -d
+
+          To access your new Mattermost deploy, navigate to ``http://<YOUR_MM_DOMAIN>:8065/`` in your browser.
```
Comment (justinegeffen) [inline source/guides/deployment.rst:330]:
```suggestion
          To access your new Mattermost deployment, navigate to ``http://<YOUR_MM_DOMAIN>:8065/`` in your browser.
```

File: source/guides/deployment.rst [hunk 6]
```diff
@@ -3,24 +3,369 @@ Deploy Mattermost
...
+
+        2. Install the Omnibus package.
+
+          .. code-block:: none
+
+            sudo apt install mattermost-omnibus -y
+
+        To issue the certificate, the installer requests a domain name and an email address from you. These are used to generate the certificate and deliver any related communications. After all the packages are installed, Omnibus runs ansible scripts that configure all the platform components and starts the server. 
+
+        3. Open a browser, navigate to your Mattermost domain by domain name (e.g. ``mymattermostserver.com``), or by the server's IP address if you're not using a domain name.
```
Comment (justinegeffen) [inline source/guides/deployment.rst:238]:
```suggestion
        3. Open a browser and navigate to your Mattermost domain either by domain name (e.g. ``mymattermostserver.com``), or by the server's IP address if you're not using a domain name. 
```

File: source/guides/deployment.rst [hunk 7]
```diff
@@ -3,24 +3,369 @@ Deploy Mattermost
...
+    
+            You should see the HTML that's returned by the Mattermost server. Note: in case firewall is used, external requests to port 8065 may be blocked. Use ``sudo ufw allow 8065`` to open port 8065.
+  
+          g. Set Mattermost to start on machine start up.
+
+            .. code:: bash
+            
+                sudo systemctl enable mattermost.service
+
+        Once you're Mattermost server is up and running, create your first Mattermost user, `invite more users <https://docs.mattermost.com/channels/manage-channel-members.html>`__, and explore the Mattermost platform. See the `Configuration Settings <https://docs.mattermost.com/configure/configuration-settings.html>`__ documentation to customize your production deployment.
```
Comment (justinegeffen) [inline source/guides/deployment.rst:216]:
```suggestion
        Once you're Mattermost server is up and running, create your first Mattermost user, `invite more users <https://docs.mattermost.com/channels/manage-channel-members.html>`__, and explore the Mattermost platform. See the `configuration settings <https://docs.mattermost.com/configure/configuration-settings.html>`__ documentation to customize your production deployment.
```

File: source/guides/deployment.rst [hunk 8]
```diff
@@ -3,24 +3,369 @@ Deploy Mattermost
...
+            
+                sudo systemctl start mattermost.service
+  
+          f. Verify that Mattermost is running.
+    
+            .. code:: bash
+            
+                curl http://localhost:8065
+    
+            You should see the HTML that's returned by the Mattermost server. Note: in case firewall is used, external requests to port 8065 may be blocked. Use ``sudo ufw allow 8065`` to open port 8065.
```
Comment (justinegeffen) [inline source/guides/deployment.rst:208]: Note formatting needs to be .rst format.

File: source/guides/deployment.rst [hunk 9]
```diff
@@ -3,24 +3,369 @@ Deploy Mattermost
...
+                sudo chmod -R g+w /opt/mattermost
+
+        7. Set up the database driver in the file ``/opt/mattermost/config/config.json``. Open the file in a text editor and make the following changes:
+  
+           **If you are using PostgreSQL:**
+
+            Set ``"DriverName"`` to ``"postgres"``
+            Set ``"DataSource"`` to the following value, replacing ``<mmuser-password>``  and ``<host-name-or-IP>`` with the appropriate values: ``"postgres://mmuser:<mmuser-password>@<host-name-or-IP>:5432/mattermost?sslmode=disable&connect_timeout=10",``
+  
+           **If you are using MySQL:**
```
Comment (justinegeffen) [inline source/guides/deployment.rst:119]:
```suggestion
           **If you're using MySQL:**
```

**Review by justinegeffen — COMMENTED** [review]
Some formatting suggestions and casing changes.

File: source/install/troubleshooting.rst (hunk 4 above)
Comment (azigler) [reply source/install/troubleshooting.rst:39]: On #5517, I proposed the troubleshooting tip differently. This should say:

> If deploying on an M1 Mac and encountering permission issues in the Docker container, redo the third step and skip this command:
> 
> `sudo chown -R 2000:2000 ./volumes/app/mattermost`

On M1, this permission change causes the deploy to not work. It's not something you can skip with a y/n/skip, but rather you omit it.

**Comment by BenLloydPearson** [comment]
@cwarnermm Andrew, Neil and I had a discussion about this and we would like to order things in the order Andrew indicated. E.g. Docker > Omnibus > Tar. We also want to refer to omnibus as the Ubuntu method because that is a much more recognizable term than omnibus. I made these changes in my last 2 PRs.

File: source/install/common-prod-deploy-docker.rst [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+                CERT_PATH=./volumes/web/cert/cert.pem
+                KEY_PATH=./volumes/web/cert/key-no-password.pem
+
+      5. Configure SSO with GitLab *(optional)*. If you want to use SSO with GitLab, and you're using a self-signed certificate, you have to add the PKI chain for your authority. This is required to avoid the ``Token request failed: certificate signed by unknown authority`` error.
+      
+            To add the PKI chain, uncomment this line in your ``.env`` file, and ensure it points to your ``pki_chain.pem`` file:
+
+            .. code:: bash
+  
+                S#GITLAB_PKI_CHAIN_PATH=<path_to_your_gitlab_pki>/pki_chain.pem
```
Comment (azigler) [inline source/install/common-prod-deploy-docker.rst:65]: Line starts with S, should be removed.

File: source/guides/deployment.rst [hunk 11]
```diff
@@ -3,24 +3,61 @@ Deploy Mattermost
...
+
+.. tip::
+
+  * See the `configuration settings <https://docs.mattermost.com/configure/configuration-settings.html>`__ documentation to learn more about customizing your production deployment.
+  
+  * Encountering issues with your deployment? See the `Deployment Troubleshooting <https://docs.mattermost.com/install/troubleshooting.html#deployment-troubleshooting>`__ documentation for details.
+
+.. tabs::
+
+    .. tab:: Using Docker
```
Comment (azigler) [inline source/guides/deployment.rst:34]: Should we change these to Docker/Ubuntu/Tar like we discussed?

**Review by azigler — COMMENTED** [review]
The Docker instructions look good to me, just submitted two minor feedback points.

**Pull request merged**